### PR TITLE
issues #190 #208 #241

### LINKED ISSUE #190: Wierd text for the tomb of sargeras quest
image

### LINKED ISSUE #208: Draktharon keep completion text wrong
image

### LINKED ISSUE #241: Stormwind quest requirement says town hall to townhal
image

Should be a castle

### PR DESCRIPTION
**Pull request by TracyTheBoy**
Wierd text for the tomb of sargeras quest #190
Draktharon keep completion text wrong #208
Stormwind quest requirement says town hall to townhal #241

Closes #190 
Closes #208 
Closes #241

### CODE REVIEW

File: src/MacroTools/QuestSystem/UtilityStructs/ObjectiveHeroWithLevelReachRect.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public ObjectiveHeroWithLevelReachRect(int targetLevel, Rectangle targetRect, st
       _targetRect = targetRect.Rect;
       _targetLevel = targetLevel;
       region target = RectToRegion(_targetRect);
-      Description = $"Any hero reaches {rectName}";
+      Description = $"Any hero that is atleast level {targetLevel} reaches {rectName}";
```
Comment (YakaryBovine) [inline src/MacroTools/QuestSystem/UtilityStructs/ObjectiveHeroWithLevelReachRect.cs:19]:
```suggestion
      Description = $"Any level {targetLevel}+ hero reaches {rectName}";
```
This is a bit shorter and we have very limited space at times.

File: src/WarcraftLegacies.Source/Quests/Stormwind/QuestStormwindCity.cs [hunk 2]
```diff
@@ -1,62 +1,71 @@
...
+
+      if (GetLocalPlayer() == completingFaction.Player) 
+        PlayThematicMusic("war3mapImported\\StormwindTheme.mp3");
     }
 
+    /// <inheritdoc />
     protected override void OnAdd(Faction whichFaction)
     {
-      whichFaction.ModObjectLimit(QuestResearchId, 1);
+      whichFaction.ModObjectLimit(ResearchId, 1);
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Stormwind/QuestStormwindCity.cs:68]: You can actually delete this entire method, as the function is already called by virtue of it being set as ResearchId.

File: src/WarcraftLegacies.Source/Quests/QuestTombOfSargeras.cs [hunk 3]
```diff
@@ -26,18 +30,15 @@ public sealed class QuestTombOfSargeras : QuestData
...
+      Global = true;
     }
     
     /// <inheritdoc />
     protected override string RewardDescription => "The Tomb of Sargeras opens";
 
-    //Todo; it would be really cool if this depended on the hero that opens it
     /// <inheritdoc />
-    protected override string CompletionPopup => "The Tomb of Sargeras has been opened.";
+    protected override string CompletionPopup => $"The Tomb of Sargeras has been opened by {GetTriggerUnit().GetProperName()}";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/QuestTombOfSargeras.cs:41]: This is brilliant, but is also a slightly risky way to do it. We cannot guarantee that the last Objective completed is ObjectiveHeroWithLevelReachRect, so we might not have a valid TriggerUnit(). 
Instead I would suggest modifying ObjectiveHeroWithLevelReachRect to set a public property on itself whenever it gets completed, then use that public property here.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Brilliant stuff as always.

File: src/WarcraftLegacies.Source/Quests/QuestTombOfSargeras.cs (hunk 3 above)
Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Quests/QuestTombOfSargeras.cs]: I hope I solved that the way you wanted to.
I added `public string EnteredRegionUnitName { get; private set; } = "an unknown hero.";` inside the `ObjectiveHeroWithLevelReachRect` and then assigned that in the `private void OnRegionEnter()` 
I then added a private field to `QuestTombOfSargeras` which stores the objective so I can use the name in the `CompletionPopup`

**Review by YakaryBovine — APPROVED** [review]
Excellent!

**Pull request merged**